rust/uefi/linux-bootloader/src/companions.rs

```diff
@@ -0,0 +1,158 @@
+use crate::cpio::{pack_cpio, Cpio};
+use alloc::{string::ToString, vec::Vec};
+use uefi::{
+    cstr16,
+    fs::{Path, PathBuf},
+    prelude::BootServices,
+    proto::device_path::{
+        text::{AllowShortcuts, DisplayOnly},
+        DevicePath,
+    },
+    CString16,
+};
+
+/// Locate files with ASCII filenames and matching the suffix passed as a parameter.
+/// Returns a list of their paths.
+pub fn find_files(
+    fs: &mut uefi::fs::FileSystem,
+    search_path: &Path,
+    suffix: &str,
+) -> uefi::Result<Vec<PathBuf>> {
+    let mut results = Vec::new();
+
+    for maybe_entry in fs.read_dir(search_path).unwrap() {
```

Comment:
So re-encoding the FileSystemError into `uefi::Result` is quite difficult because the enum is Io, Path or Utf8Encoding and it's probably going to be very unhelpful to eat the low level errors into high level errors that will produce difficult to debug code IMHO. A better solution would be to have ways to propagate the original error up but there's impedence mismatch with our error management here.